storage/sql/coresql/wrapper/wrapper.go

```diff
@@ -0,0 +1,108 @@
+// Copyright 2017 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package wrapper defines a wrapping mechanism for databases and decouples the logic
+// of accessing the database from the creation of statements in a specific SQL format.
+// It's strongly related to the coresql package but cannot be packaged with it because it's
+// then impossible to avoid import cycles.
+package wrapper
+
+import (
+	"context"
+	"database/sql"
+)
+
+// TreeStatementProvider provides SQL statement objects for raw tree storage.
+type TreeStatementProvider interface {
```

Comment:
Currently there's a somewhat undocumented contract between the wrapper and coresql packages in that some of the `Stmt` types returned here require positional parameters in a particular ordering, I'm wondering if we can refactor that a bit so that wrapperDBs execute the statements themselves after being passed in the params, but in the meantime adding some coments on these interface methods detailing what the expected params & ordering are for the returned `Stmt` would help make the contract clearer.
WDYT?